File: MissingNumber.js
class NumberUtils {
    static getMaxValue(array) {
      return Math.max(...array);
    }
  
    static findMissingNumber(array) {
      const max = this.getMaxValue(array);
      for (let i = 0; i <= max; i++) {
        if (!array.includes(i)) {
          return i;
        }
      }
      return null; // If nothing is missing
    }
  }
  
  class MissingNumberService {
    constructor(numberArray) {
      this.numberArray = numberArray;
    }
  
    getMissingNumber() {
      return NumberUtils.findMissingNumber(this.numberArray);
    }
  }
  
  // Usage
  const nums = [9, 6, 4, 2, 3, 5, 7, 0, 1, 8, 11];
  const missingService = new MissingNumberService(nums);
  console.log(missingService.getMissingNumber());

File: MissingNumber.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class NumberUtils
{
    public static int GetMaxValue(List<int> array)
    {
        return array.Max();
    }

    public static int? FindMissingNumber(List<int> array)
    {
        int max = GetMaxValue(array);
        for (int i = 0; i <= max; i++)
        {
            if (!array.Contains(i))
            {
                return i;
            }
        }
        return null; // If nothing is missing
    }
}

public class MissingNumberService
{
    private List<int> _numberArray;

    public MissingNumberService(List<int> numberArray)
    {
        _numberArray = numberArray;
    }

    public int? GetMissingNumber()
    {
        return NumberUtils.FindMissingNumber(_numberArray);
    }
}

public class Program
{
    public static void Main()
    {
        var nums = new List<int> { 9, 6, 4, 2, 3, 5, 7, 0, 1, 8, 11 };
        var missingService = new MissingNumberService(nums);
        var missingNumber = missingService.GetMissingNumber();

        Console.WriteLine(missingNumber.HasValue ? $"Missing number: {missingNumber}" : "No missing number found");
    }
}
